airflow/dag_processing/manager.py

```diff
@@ -547,11 +546,13 @@ def deactivate_stale_dags(
                 to_deactivate.add(dag.dag_id)
 
         if to_deactivate:
-            deactivated = (
-                session.query(DagModel)
-                .filter(DagModel.dag_id.in_(to_deactivate))
-                .update({DagModel.is_active: False}, synchronize_session="fetch")
+            deactivated_dagmodel = session.execute(
+                update(DagModel)
+                .where(DagModel.dag_id.in_(to_deactivate))
+                .values(is_active=False)
+                .execution_options(synchronize_session="fetch")
             )
+            deactivated = len([dagmodel for dagmodel, in deactivated_dagmodel])
```

Comment:
I wonder if we can just get the number of affected rows; something like `deactivated_dagmodel.rowcount` instead of unnecessary `fetch`

Reply:
Yes, please use `.rowcount` here. There are various examples if you search for this.

Reply:
made the change